### ISSUE TITLE
The system cannot find the path specified. / Unison not running in VM

### ISSUE THREAD

**Issue by pbowyer** [dcosson/vagrant-unison#16]
The speedup of using Unison instead of NFS is awesome (I have just got it installed :+1: ) but I'm getting this message all the time throughout the polling. Host is Windows 7 Pro, Guest is Ubuntu 14.04.

```
λ vagrant unison-sync-polling
Unisoning changes from {host}::D:/ia/ --> {guest VM}::/var/www
Running unison D:/ia/ ssh://vagrant@127.0.0.1//var/www -terse -repeat 1 -sshargs "-p 2222  -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -i D:/ia/.vagrant/machines/iadev/virtualbox/private_key"
Warning: Permanently added '[127.0.0.1]:2222' (RSA) to the list of known hosts.
The system cannot find the path specified.
The system cannot find the path specified.
Unison not running in VM
changed  ---->            .idea/workspace.xml
changed  ---->            Vagrantfile
[BGN] Updating file .idea/workspace.xml from D:/ia to //iadev//var/www
[BGN] Updating file Vagrantfile from D:/ia to //iadev//var/www
[END] Updating file .idea/workspace.xml
[END] Updating file Vagrantfile
Synchronization complete at 16:34:05  (2 items transferred, 0 skipped, 0 failed)
The system cannot find the path specified.
The system cannot find the path specified.
Unison not running in VM
The system cannot find the path specified.
The system cannot find the path specified.
Unison not running in VM
         <---- new file   test.txt
[BGN] Copying test.txt from //iadev//var/www to D:/ia
Shortcut: copied D:/ia/test.txt from local file D:/ia/web/vendor/phpforce/soap-client/tests/Phpforce/SoapClient/Tests/AbstractResultTest.php
[END] Copying test.txt
Synchronization complete at 16:34:31  (1 item transferred, 0 skipped, 0 failed)
The system cannot find the path specified.
The system cannot find the path specified.
Unison not running in VM
The system cannot find the path specified.
The system cannot find the path specified.
Unison not running in VM
```

What does it mean, and is there any option to turn on a more verbose output?

**Comment by dcosson** [comment]
Sorry for the delayed response

Could it be a combination of windows & unix paths? For instance, shouldn't all the windows paths be using backslashes? 

Try running the unison command explicitly in your host, i.e:

```
unison D:/ia/ ssh://vagrant@127.0.0.1//var/www -terse -repeat 1 -sshargs "-p 2222  -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -i D:/ia/.vagrant/machines/iadev/virtualbox/private_key
```

If that still shows the error, i would try removing the /dev/null part since i don't think that makes sense on windows, and double check that the private_key path is correct, maybe it needs to be changed to use backslashes?

I don't have a windows host to test this on (or own a windows license to install as a vagrant guest) so unfortunately I can't be much help in debugging this.

**Comment by denblackstache** [comment]
@dcosson Yeah, with windows paths specified to use backslashes there is no such error about hosts.

> Warning: Permanently added '[127.0.0.1]:2222' (RSA) to the list of known hosts.

Also I cannot reproduce this subject error after that manipulation. Maybe it also fixes that.

**Comment by dcosson** [comment]
> Also I cannot reproduce thid subject error after that manipulation. Maybe it also fixes that.

@Hyperc0der Not sure I follow, sorry. Are you referring to the ssh warning quoted? That's because of the ssh settings with user known hosts being set to /dev/null, if you have a solution that'll let us not skip ssh authentication but will continue to work if you destroy & re-create the vm I'd love to fix this.

**Comment by dcosson** [comment]
@pbowyer If you got this working and want to submit a patch that'll use the right paths on Windows or OSX, that would be awesome. I don't ever use a Windows host.

**Comment by pbowyer** [comment]
@dcosson I got it working and will have to dig out this VM to see how (haven't used it for a few months). I had it working well for our 2-person team (one a Windows host, the other OSX)

**Comment by mmmpop** [comment]
According to the Unison docs, version 2.4.8 introduced a new behavior that causes it to bail when '-watch' is set in a way it cannot handle. I haven't investigated much more than that, but even with this intermittent message in my CLI, the sync works flawlessly! Thanks for the hard work on this and perhaps I'll get around to a PR when I get a bit of free time.

**Comment by mmmpop** [comment]
According to the Unison docs, version 2.48.4 introduced a new behavior:

> Do not restart Unison in case of uncaught exception when the repeat preference is set. This seems safer. And it does not work, for instance, in case of lost connection.

I haven't investigated much more than that, but even with this intermittent message in my CLI, the sync works flawlessly! Thanks for the hard work on this and perhaps I'll get around to a PR when I get a bit of free time.
